base/src/main/java/org/aya/tyck/ExprTycker.java

```diff
@@ -434,8 +434,9 @@ yield switch (term.normalize(state, NormalizeMode.WHNF)) {
               yield fail(lam, term, new CubicalProblem.DimensionMismatch(lam, cubeParams.size(), plam._1.size()));
             // we allow lambda params to be typed explicitly --- check them against I.
             var params = plam._1.map(p -> {
-              inherit(p.type(), FormTerm.Interval.INSTANCE);
-              return new Term.Param(p, FormTerm.Interval.INSTANCE);
+              var i = synthesize(p.type());
+              unifyTyReported(FormTerm.Interval.INSTANCE, i.wellTyped, p.type());
```

Comment:
Isn't this the same thing?

Reply:
No. `inherit(p.type(), I)` says `p.type() : I` while we expect `p.type() = I`